### ISSUE TITLE
Firefox doesn't support comments in provided async function

### ISSUE THREAD

**Issue by JamesLMilner** [developit/greenlet#32]
More just an observation than a bug; Firefox doesn't appear to handle comments inside of the passed async function. This appears to be because it puts all the code onto one line which causes the parsing to break via a SyntaxError i.e.:

```
SyntaxError: missing } after function body
data:$$=async%20(data,%20stringify)%20=>%20{//%20Commentif%20(stringify)%20{data%20=%20JSON.parse(da:1:416
note: { opened at line 1, column 30
```

in short, this will cause an error:

```javascript
	const exampleAsyncFunc = async (data) => {
            // A wild comment appears
 	    return data
	}
```

but this does not:

```javascript
	const exampleAsyncFunc = async (data) => {
 	    return data
	}
```

**Comment by developit** [comment]
Looks like the fix here is to just manually encode the URI instead of letting the browser do it.
We could potentially even just apply it to `asyncFunction` here:
https://github.com/developit/greenlet/blob/d8467e0f45049dd71ee899dd73093d5137e41435/greenlet.js#L15

**Comment by developit** [comment]
Update: if we switch back to Blob URLs to address the CSP issues, this error goes away.

**Comment by developit** [comment]
This is fixed by #38.
